radix-engine-common/src/math/decimal.rs

```diff
@@ -430,7 +504,107 @@ macro_rules! impl_arith_ops {
             type Output = Decimal;
 
             fn safe_div(self, other: Decimal) -> Option<Self::Output> {
-                Decimal::from(self).safe_div(other)
+                Decimal::try_from(self).ok()?.safe_div(other)
+            }
+        }
+
+        impl Add<$type> for Decimal {
+            type Output = Self;
+
+            #[inline]
+            fn add(self, other: $type) -> Self::Output {
+                self.safe_add(other).expect("Overflow")
```

Comment:
As agreed this one and all other unsafe operations are implemented by wrapping the safe operations and unwrapping the result. This  might slightly impact the performance.